vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.cc

```diff
@@ -0,0 +1,126 @@
+
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events.h"
+
+#include <algorithm>
+#include <vector>
+
+#include "base/check.h"
+#include "bat/ads/internal/base/logging_util.h"
+#include "bat/ads/internal/base/search_engine/search_engine_results_page_util.h"
+#include "bat/ads/internal/base/search_engine/search_engine_util.h"
+#include "bat/ads/internal/features/text_embedding_features.h"
+#include "bat/ads/internal/locale/locale_manager.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_data.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_processing.h"
+#include "bat/ads/internal/resources/contextual/text_embedding/text_embedding_resource.h"
+#include "bat/ads/internal/resources/language_components.h"
+#include "bat/ads/internal/resources/resource_manager.h"
+#include "bat/ads/internal/tabs/tab_manager.h"
+#include "url/gurl.h"
+
+namespace ads {
+namespace processor {
+
+TextEmbedding::TextEmbedding(resource::TextEmbedding* resource)
+    : resource_(resource) {
+  DCHECK(resource_);
+
+  LocaleManager::GetInstance()->AddObserver(this);
+  ResourceManager::GetInstance()->AddObserver(this);
+  TabManager::GetInstance()->AddObserver(this);
+}
+
+TextEmbedding::~TextEmbedding() {
+  LocaleManager::GetInstance()->RemoveObserver(this);
+  ResourceManager::GetInstance()->RemoveObserver(this);
+  TabManager::GetInstance()->RemoveObserver(this);
+}
+
+bool TextEmbedding::IsEmbeddingEnabled() {
+  return targeting::features::IsTextEmbeddingEnabled();
+}
+
+void TextEmbedding::Process(const std::string& text) {
+  if (!resource_->IsInitialized()) {
+    BLOG(1, "Failed to process token embeddings as resource not initialized");
+    return;
+  }
+
+  ml::pipeline::EmbeddingProcessing* embedding_proc_pipeline = resource_->Get();
+
+  const std::string cleaned_text =
+      embedding_proc_pipeline->CleanText(text, true);
+  if (cleaned_text.length() == 0) {
+    BLOG(1, "No text available for embedding");
+    return;
+  }
+
+  ml::pipeline::TextEmbeddingData embedding_data =
+      embedding_proc_pipeline->EmbedText(cleaned_text);
+  if (embedding_data.embedding.GetNonZeroElementsCount() == 0) {
+    BLOG(1, "Text not embedded");
+    return;
+  }
+
+  const std::string embedding_formatted =
+      embedding_data.embedding.GetVectorAsString();
+  BLOG(1, "Embedding: " << embedding_formatted);
+  LogTextEmbeddingHTMLEvent(
+      embedding_formatted, embedding_data.text_hashed, [](const bool success) {
+        if (!success) {
+          BLOG(1, "Failed to text embedding html event");
+          return;
+        }
+
+        PurgeStaleTextEmbeddingHTMLEvents([](const bool success) {
+          if (!success) {
+            BLOG(1, "Failed to purge stale text embedding html events");
+            return;
+          }
+        });
+
+        GetTextEmbeddingEventsFromDatabase();
```

Comment:
We don't need to get the events after we logged a new event, right?

Reply:
Perhaps this can be conditional on logging verbosity? Otherwise, the function is never used and should potentially not be a part of this PR (earlier comment mentioned not including unused functions in PRs)

Reply:
I would just remove it here, or maybe create another function that logs embeddings, or even pass a printing callback to the Getter function as discussed in the other comment. Re the getter, I think it would still be useful to have as it completes the functionality of the PR.

Reply:
Removed here. Added as as part of a unit test in `database_migration_unittest.cc`